packages/slice-machine/lib/builders/CustomTypeBuilder/index.tsx

```diff
@@ -1,28 +1,143 @@
-import { Box } from "theme-ui";
+import { DropdownMenuItem, Icon } from "@prismicio/editor-ui";
+import { type FC, useState } from "react";
 
+import type { CustomTypeSM } from "@lib/models/common/CustomType";
+import {
+  Window,
+  WindowFrame,
+  WindowTabs,
+  WindowTabsContent,
+  WindowTabsList,
+  WindowTabsTrigger,
+} from "@src/components/Window";
+import useSliceMachineActions from "@src/modules/useSliceMachineActions";
+
+import CreateModal from "./TabModal/create";
+import DeleteModal from "./TabModal/delete";
+import UpdateModal from "./TabModal/update";
 import TabZone from "./TabZone";
-import Tabs from "./Layout";
-import { CustomTypeSM } from "@lib/models/common/CustomType";
 
-type CustomTypeBuilderProps = {
-  customType: CustomTypeSM;
-};
-const CustomTypeBuilder = ({ customType }: CustomTypeBuilderProps) => {
+type CustomTypeBuilderProps = { customType: CustomTypeSM };
+
+type DialogState =
+  | { type: "CREATE_CUSTOM_TYPE" }
+  | { type: "UPDATE_CUSTOM_TYPE"; tabKey: string }
+  | { type: "DELETE_CUSTOM_TYPE"; tabKey: string }
+  | undefined;
+
+export const CustomTypeBuilder: FC<CustomTypeBuilderProps> = (props) => {
+  const { customType } = props;
+
+  const [tabValue, setTabValue] = useState(customType.tabs[0]?.key);
+
+  const [dialog, setDialog] = useState<DialogState>();
+  const { createCustomTypeTab, updateCustomTypeTab, deleteCustomTypeTab } =
+    useSliceMachineActions();
+
   return (
-    <Tabs
-      tabs={customType.tabs}
-      renderTab={({ value, sliceZone, key }) => (
-        <Box sx={{ mt: 4 }}>
-          <TabZone
-            fields={value}
-            sliceZone={sliceZone}
-            tabId={key}
-            customType={customType}
-          />
-        </Box>
-      )}
-    />
+    <>
+      <Window>
+        {customType.format === "page" ? <WindowFrame /> : undefined}
+        <WindowTabs onValueChange={setTabValue} value={tabValue}>
+          <WindowTabsList
+            onAddNewTab={() => {
+              setDialog({ type: "CREATE_CUSTOM_TYPE" });
+            }}
+          >
+            {customType.tabs.map((tab) => (
+              <WindowTabsTrigger
+                key={tab.key}
+                menu={
+                  <>
+                    <DropdownMenuItem
+                      onSelect={() => {
+                        setDialog({
+                          type: "UPDATE_CUSTOM_TYPE",
+                          tabKey: tab.key,
+                        });
+                      }}
+                      startIcon={<Icon name="edit" />}
+                    >
+                      Rename
+                    </DropdownMenuItem>
+                    <DropdownMenuItem
+                      color="tomato"
+                      disabled={customType.tabs.length < 2}
```

Comment:
ℹ️ Just so you know, personally I prefer these conditions:

- `customType.tabs.length <= 1`
- `customType.tabs.length === 1`

Just because when I read 2, I'm always like what what, why 2 haha